swift/llm/utils/template.py

```diff
@@ -1008,13 +1008,19 @@ def data_collator(self, batch: List[Dict[str, Any]], padding_to: Optional[int] =
         if 'inputs_embeds' in batch[0]:
             inputs_embeds = [b['inputs_embeds'] for b in batch]
             res['inputs_embeds'] = inputs_embeds
-            res['attention_mask'] = [
-                torch.ones((inputs_embeds[i].shape[0]), dtype=torch.int64) for i in range(len(inputs_embeds))
-            ]
+            if 'attention_mask' in batch[0]:
+                res['attention_mask'] = batch[0]['attention_mask']
```

Comment:
The batch size may be greater than 1.

Reply:
Okay, let me think about how to deal with batch size greater than 1.